Subject:  Script changes

Patchset changes:
1.  Add shebang to all bash scripts.
2.  Make script runnable from any directory when invoking with absolute path.
3.  Add new environment variables:
3a. C2TCP_ELEVATE_CMD -- sudo, pkexec, " "
3b. C2TCP_TARGET_DELAY -- allow to override target delay in milliseconds
3c. C2TCP_BROWSER -- allow to change the web browser command or disable browser.

diff -urpN c2tcp-2.0_p9999.orig/build.sh c2tcp-2.0_p9999/build.sh
--- c2tcp-2.0_p9999.orig/build.sh	2023-02-16 22:10:28.073464055 -0800
+++ c2tcp-2.0_p9999/build.sh	2023-02-16 22:11:52.561466839 -0800
@@ -1,3 +1,5 @@
+#!/bin/bash
+cd $(dirname "$0")
 g++ -pthread src/server-standalone.cc src/flow.cc -o server
 g++ src/client.c -o client
 
diff -urpN c2tcp-2.0_p9999.orig/run-sample.sh c2tcp-2.0_p9999/run-sample.sh
--- c2tcp-2.0_p9999.orig/run-sample.sh	2023-02-16 22:10:40.721464472 -0800
+++ c2tcp-2.0_p9999/run-sample.sh	2023-02-16 22:16:40.245476319 -0800
@@ -1,3 +1,7 @@
+#!/bin/bash
+cd $(dirname "$0")
+C2TCP_ELEVATE_CMD=${C2TCP_ELEVATE_CMD:-sudo} # accepts:  sudo, pkexec, " "
+C2TCP_BROWSER=${C2TCP_BROWSER:-"firefox"}
 source setup.sh
 qsize=100
 down="downlink-4g-no-cross-subway.pps"
@@ -9,25 +13,36 @@ init_alpha=150
 user=`whoami`
 anal_script="mm-thr"
 
-target=0    #Application's desired Target delay: "0" means Target = 2 x minRTT
+target=${C2TCP_TARGET_DELAY:-0}    #Application's desired Target delay: "0" means Target = 2 x minRTT
 #target=50
 delay=10    #Unidirectioanl delay: 10ms
 anal_script="mm-throughput-graph-modified"
-cmd="sudo ./client \$MAHIMAHI_BASE 1 $port & sleep $duration && killall client"
+cmd="${C2TCP_ELEVATE_CMD} ./client \$MAHIMAHI_BASE 1 $port & sleep $duration && killall client"
 
 downlink=${down}
 uplink=wired48
 trace_="$uplink-$downlink"
 log="$queue-$delay-$trace_-$qsize-$target-$duration"
 
-sudo ./server $port $target $init_alpha $tuning_period &
+${C2TCP_ELEVATE_CMD} ./server $port $target $init_alpha $tuning_period &
 
-sudo -u $user mm-delay $delay mm-link traces/$uplink traces/$downlink --uplink-log=log/up-mul-$log --downlink-log=log/down-mul-$log --uplink-queue=droptail --uplink-queue-args=\"packets=$qsize\" --downlink-queue=droptail --downlink-queue-args=\"packets=$qsize\"  -- sh -c "$cmd"
+elevate_arg=""
+if [[ "${C2TCP_ELEVATE_CMD}" == "sudo" ]] ;
+    elevate_arg="-u"
+elif [[ "${C2TCP_ELEVATE_CMD}" == "pkexec" ]] ;
+    elevate_arg="--user"
+fi
 
-sudo killall server client
+${C2TCP_ELEVATE_CMD} ${elevate_arg} $user mm-delay $delay mm-link traces/$uplink traces/$downlink --uplink-log=log/up-mul-$log --downlink-log=log/down-mul-$log --uplink-queue=droptail --uplink-queue-args=\"packets=$qsize\" --downlink-queue=droptail --downlink-queue-args=\"packets=$qsize\"  -- sh -c "$cmd"
+
+${C2TCP_ELEVATE_CMD} killall server client
 echo $log >>log/summary.tr
 ./$anal_script 500 log/down-mul-$log 1>log/fig-$log.svg 2>>log/summary.tr
 echo "---------------------------------------">>log/summary.tr
 cat log/summary.tr
-firefox log/fig-$log.svg
+if [[ "${C2TCP_BROWSER}" =~ ("none"|"skip"|"disable") || "${C2TCP_BROWSER}" == " " || -z "${C2TCP_BROWSER}" ]] ; then
+    :;
+elif [[ -n "${C2TCP_BROWSER}" ]] ; then
+    ${C2TCP_BROWSER} log/fig-$log.svg
+fi
 
diff -urpN c2tcp-2.0_p9999.orig/setup.sh c2tcp-2.0_p9999/setup.sh
--- c2tcp-2.0_p9999.orig/setup.sh	2023-02-16 22:10:40.721464472 -0800
+++ c2tcp-2.0_p9999/setup.sh	2023-02-16 22:11:52.597466840 -0800
@@ -1,8 +1,11 @@
-sudo sysctl -q net.ipv4.tcp_wmem="4096 32768 4194304" #Doubling the default value from 16384 to 32768
-sudo sysctl -w -q net.ipv4.tcp_low_latency=1
-sudo sysctl -w -q net.ipv4.tcp_autocorking=0
-sudo sysctl -w -q net.ipv4.tcp_no_metrics_save=1
-sudo sysctl -w -q net.ipv4.ip_forward=1
+#!/bin/bash
+cd $(dirname "$0")
+C2TCP_ELEVATE_CMD=${C2TCP_ELEVATE_CMD:-sudo} # accepts:  sudo, pkexec, " "
+${C2TCP_ELEVATE_CMD} sysctl -q net.ipv4.tcp_wmem="4096 32768 4194304" #Doubling the default value from 16384 to 32768
+${C2TCP_ELEVATE_CMD} sysctl -w -q net.ipv4.tcp_low_latency=1
+${C2TCP_ELEVATE_CMD} sysctl -w -q net.ipv4.tcp_autocorking=0
+${C2TCP_ELEVATE_CMD} sysctl -w -q net.ipv4.tcp_no_metrics_save=1
+${C2TCP_ELEVATE_CMD} sysctl -w -q net.ipv4.ip_forward=1
 
 DOWNLINKS=("downlink-4g-no-cross-subway.pps" "downlink-4g-with-cross-subway.pps" "downlink-3g-no-cross-subway.pps" "downlink-4g-no-cross-times" "downlink-4g-with-cross-times" "downlink-3g-with-cross-times-1" "downlink-3g-no-cross-times-1" "ATT-LTE-driving-2016.down" "ATT-LTE-driving.down"  "TMobile-LTE-driving.down" "TMobile-UMTS-driving.down" "Verizon-EVDO-driving.down" "Verizon-LTE-driving.down");
 UPLINKS=("wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48" "wired48");
